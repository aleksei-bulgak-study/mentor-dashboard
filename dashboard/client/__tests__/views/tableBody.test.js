import React from 'react';
import { shallow } from 'enzyme';
import TableBody from '../../src/views/tableBody';

describe('TableBody', () => {
  it('default snapshot', () => {
    // given
    const props = {
      tasks: [
        { name: 'test', status: 'Checking', link: 'https://example.com/test' },
        { name: 'test2', status: 'In Progress', link: 'https://example.com/test2' },
      ],
      students: ['student1', 'student2'],
      results: [
        {
          github: 'https://github.com/student1',
          tasks: {
            test: { mark: 100, pr: 'https://example.com/task' },
            test2: { mark: 30, pr: 'https://example.com/task2' },
          },
        },
        {
          github: 'https://github.com/student2',
          tasks: {
            test2: { mark: 10, pr: 'https://example.com/task2' },
          },
        },
      ],
    };
    // when
    const wrapper = shallow(<TableBody {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });
});
