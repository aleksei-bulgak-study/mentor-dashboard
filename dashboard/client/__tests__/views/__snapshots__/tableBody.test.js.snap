// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody default snapshot 1`] = `
<tbody>
  <TableRow
    key="test"
    results={
      Array [
        Object {
          "github": "https://github.com/student1",
          "tasks": Object {
            "test": Object {
              "mark": 100,
              "pr": "https://example.com/task",
            },
            "test2": Object {
              "mark": 30,
              "pr": "https://example.com/task2",
            },
          },
        },
        Object {
          "github": "https://github.com/student2",
          "tasks": Object {
            "test2": Object {
              "mark": 10,
              "pr": "https://example.com/task2",
            },
          },
        },
      ]
    }
    students={
      Array [
        "student1",
        "student2",
      ]
    }
    task={
      Object {
        "link": "https://example.com/test",
        "name": "test",
        "status": "Checking",
      }
    }
  />
  <TableRow
    key="test2"
    results={
      Array [
        Object {
          "github": "https://github.com/student1",
          "tasks": Object {
            "test": Object {
              "mark": 100,
              "pr": "https://example.com/task",
            },
            "test2": Object {
              "mark": 30,
              "pr": "https://example.com/task2",
            },
          },
        },
        Object {
          "github": "https://github.com/student2",
          "tasks": Object {
            "test2": Object {
              "mark": 10,
              "pr": "https://example.com/task2",
            },
          },
        },
      ]
    }
    students={
      Array [
        "student1",
        "student2",
      ]
    }
    task={
      Object {
        "link": "https://example.com/test2",
        "name": "test2",
        "status": "In Progress",
      }
    }
  />
</tbody>
`;
